wp-content/civi-extensions/goonjcustom/api/v3/wp-content/civi-extensions/goonjcustom/api/v3/ContributionFilter.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+require_once 'api/Wrapper.php';
+
+use Civi\Api4\Campaign;
+use Civi\Api4\Generic\AbstractAction;
+
+/**
+ *
+ */
+class CRM_Goonjcustom_APIWrappers_ContributionFilter implements API_Wrapper {
+
+  private $allowAccess = TRUE;
+
+  /**
+   *
+   */
+  public function fromApiInput($apiRequest) {
+    if (!($apiRequest instanceof AbstractAction)) {
+      throw new \InvalidArgumentException('Expected Civi\Api4\Generic\AbstractAction');
+    }
+
+    $params = $apiRequest->getParams();
+    $clientIP = $_SERVER['REMOTE_ADDR'] ?? '';
+
+    // $allowedIPs = ['20.235.42.26'];
+    // $clientIP = $_SERVER['REMOTE_ADDR'] ?? '';
+
+    // if (!in_array($clientIP, $allowedIPs)) {
+    //   echo 'Access denied: Unauthorized IP address.';
+    //   exit;
+    // }
```

Comment:
This is because access has been restricted to the Goonj website’s IP only. However, I need to test it on production using Postman. So for now, I’ve commented out this restriction. Once everything works as expected, I’ll uncomment the code in the new PR.